Adding support for OCI exception mapping logic in Litellm

ORC-EMG opened the pull request:
## Adding support for OCI exception mapping logic in Litellm 


## Relevant issues

In testing LiteLLM with OCI, error mapping is not occurring as expected. Errors are returned wrapped in a LiteLLM 500 error by the LiteLLM exception_mapping_utils. py file. This includes when errors are actually 401, 404, 429, and other errors as returned by OCI they are wrapped in a 500 LiteLLM error. This can lead to misleading developers and users of the actual issue occurring.  

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🐛 Bug Fix
✅ Test

## Changes

Added exception mapping functions and logic specifically for OCI in the oci_exception_mapping_utils.py file in the same style as other providers. Added the test files in the tests directory per the contributing hard requirement checklist..

ORC-EMG committed:
Update exception_mapping_utils.py to add OCI error mapping support

ORC-EMG committed:
Update test_exception_mapping_utils.py - adding test cases for OCI exception mapping

ORC-EMG committed:
Update test_exception_mapping_utils.py updated testing utility for oci exception mapping

ORC-EMG committed:
Update test_exception_mapping_utils.py updated to comply with styling guide

ORC-EMG committed:
Update exception_mapping_utils.py - fixed styling guide violations like line length violations and inline comments.

ORC-EMG committed:
Update exceptions.py to include ConflictError class for test case related to HTTP 409 code for OCI error mapping update, added ConflictError to list of exception types

ORC-EMG committed:
Update exception_mapping_utils.py - added conflict error to the import from exceptions.py as part of test case for added logic in PR

ORC-EMG commented:
<img width="1336" height="27" alt="Screenshot 2025-12-02 at 8 19 43 PM" src="https://github.com/user-attachments/assets/5bd6b9e3-8b16-40d3-b593-4482287b2a98" />
<img width="921" height="62" alt="Screenshot 2025-12-02 at 8 18 38 PM" src="https://github.com/user-attachments/assets/3031362c-bc04-42ee-8203-64dd754c77c0" />

krrishdholakia commented on litellm/exceptions.py:
```diff
@@ -152,7 +152,50 @@ def __repr__(self):
             _message += f", LiteLLM Max Retries: {self.max_retries}"
         return _message
+        
+class ConflictError(openai.ConflictError):  # type: ignore
```
@ORC-EMG you should bump the litellm openai dep as well in that case, as we support 2.8.0 + - https://github.com/BerriAI/litellm/blob/6d831ffff4e279e859c733ed823ed7214da0ecea/pyproject.toml#L25

ORC-EMG committed:
Update pyproject.toml - updating required openai version that supports ConflictError

ORC-EMG committed:
Merge branch 'main' into litellm_oci_exception_mapping

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update exception_mapping_utils.py

Update to address Dead code: variable opc_request_id is extracted but never used

opc_request_id is extracted from the error payload (lines 1015-1017) but is never referenced in the exception mapping block — not appended to messages, not passed to exception constructors, and not logged. Either use it (e.g., append to oci_message) or remove it:

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update exception_mapping_utils.py

removed duplicate comment string

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update exception_mapping_utils.py

identation issues addressed on lines 1020-1035 to match surrounding elif block. Addressed duplicate assignment of exception_mapping_worked and exception_provider (1018-1019) by removing them.

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update test_exception_mapping_utils.py

Add authorization error to exception test cases in line 288

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update test_exception_mapping_utils.py

Add a test that passes a JSON-formatted string as the exception message (no status_code attribute) to ensure the JSON-parsing path works correctly end-to-end:

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update __init__.py

added ConflictError

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ORC-EMG committed:
Update litellm/litellm_core_utils/exception_mapping_utils.py

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>
